include/triton/Analysis/Utility.h

```diff
@@ -334,6 +337,19 @@ std::unique_ptr<DataFlowSolver> createDataFlowSolver();
 
 triton::MakeTensorPtrOp getMakeTensorPtrOp(Value v);
 
+SmallVector<unsigned, 3>
+mmaVersionToInstrShape(int version, ArrayRef<int64_t> shape, Type type);
+
+Value getParentValueWithSameEncoding(Attribute layout, Value value);
+
+inline Type getInput0EltType(Value value) {
+  auto *op = value.getDefiningOp();
+  if (auto type = op->getOperands()[0].getType().dyn_cast<RankedTensorType>()) {
```

Comment:
How do we know the value comes from an `Operation*` and this operations has an operand?